### PR TITLE
Bucket online offline scripts continuation

### PR DESCRIPTION
**Pull request by ashvindersingh**
This PR implements:
- Resync scenarios for db_online_offline
- webhook scenarios for db_online_offline

### CODE REVIEW

File: functional_tests/test_db_online_offline.py [hunk 1]
```diff
@@ -636,3 +636,42 @@ def test_multiple_dbs_unique_buckets_lose_tap(cluster, conf, num_docs):
 #     assert(status == 500)
 
 
+## Scenario 17
```
Comment (sethrosetter) [inline functional_tests/test_db_online_offline.py]: Is this a test to be resurrected for 1.3 or is this just a stale scenario?

File: functional_tests/test_db_online_offline_resync.py [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+
+# implements scenario: 11
+# With DB in online state, put a large number of docs (enough to cause _resync to run for 10-15 seconds),
+# put DB offline, run _resync, attempt to bring DB online while _resync is running,
+# expected result _online will fail with status 503, when _resync is complete,
+# attempt to bring DB _online, expected result _online will succeed, return status 200.
+@pytest.mark.sanity
+@pytest.mark.dbonlineoffline
+@pytest.mark.parametrize("num_users", [5])
+@pytest.mark.parametrize("num_docs", [100])
```
Comment (sethrosetter) [inline functional_tests/test_db_online_offline_resync.py]: Is this enough docs to cause _resync to run for 10-15 seconds?

File: functional_tests/test_db_online_offline_resync.py [hunk 3]
```diff
@@ -0,0 +1,441 @@
...
+    try:
+        async_resync_result = pool.apply_async(admin.db_resync, ("db",))
+        log.info("resync issued !!!!!!")
+    except Exception as e:
+        log.info("Catch resync exception: {}".format(e))
+
+    time.sleep(1)
+    resync_occured = False
+
+    for i in range(20):
```
Comment (sethrosetter) [inline functional_tests/test_db_online_offline_resync.py]: Should this poll untill the resync is complete?

File: functional_tests/test_db_online_offline_resync.py [hunk 4]
```diff
@@ -0,0 +1,441 @@
...
+                    assert False
+
+        time.sleep(1)
+        if resync_occured:
+            break
+
+    time.sleep(10)
+
+    status = admin.bring_db_online(db="db")
+    log.info("online request issued !!!!! response status: {}".format(status))
```
Comment (sethrosetter) [inline functional_tests/test_db_online_offline_resync.py]: Maybe assert on status?

File: functional_tests/test_db_online_offline_resync.py (hunk 2 above)
Comment (ashvindersingh) [inline functional_tests/test_db_online_offline_resync.py]: It was enough to run the resync and do the test operation within the time. Adding more docs will just run resync longer but will not have any impact on the test condition

File: functional_tests/test_db_online_offline.py (hunk 1 above)
Comment (ashvindersingh) [inline functional_tests/test_db_online_offline.py]: Yes for 1.3. depends upon the put db_config

File: lib/verify.py [hunk 5]
```diff
@@ -1,5 +1,8 @@
 
 from lib.user import User
+from lib import settings
+import logging
+log = logging.getLogger(settings.LOGGER)
```
Comment (sethrosetter) [inline lib/verify.py]: Please remove redundant imports

File: lib/admin.py [hunk 6]
```diff
@@ -5,6 +5,8 @@
 from lib.user import User
 from lib.scenarioprinter import ScenarioPrinter
 from lib import settings
+from requests.packages.urllib3.util import Retry
```
Comment (sethrosetter) [inline lib/admin.py]: What is this used for?

File: functional_tests/test_db_online_offline_resync.py [hunk 7]
```diff
@@ -0,0 +1,441 @@
...
+    in_parallel(user_objects, 'get_changes')
+
+    # every user should have same number of docs
+    # total/expected docs = num_users * num_docs
+    recieved_docs = in_parallel(user_objects, 'get_num_docs')
+
+    expected_docs = num_users * num_docs
+    for user_obj, docs in recieved_docs.items():
+        log.info('User {} got {} docs, expected docs: {}'.format(user_obj.name, docs, expected_docs))
+        assert docs == 1 + expected_docs
```
Comment (sethrosetter) [inline functional_tests/test_db_online_offline_resync.py]: Can you update get_num_docs to work with both original sync_gateway mode as well as distributed index mode. (i.e. ignore _user docs)

**Pull request merged**